pkg/api/api.go

```diff
@@ -0,0 +1,676 @@
+func (c *Client) DownloadBatch(ctx context.Context, dte *time.Time, idr string, wsk io.WriteSeeker) error {
+	return c.downloadEntity(ctx, fmt.Sprintf("batches/%s/%s/download", dte.Format(dateFormat), idr), wsk)
+}
+
+// GetSnapshots retrieves a list of all snapshots and returns an error if any.
+func (c *Client) GetSnapshots(ctx context.Context, req *Request) ([]*Snapshot, error) {
+	sps := []*Snapshot{}
+	return sps, c.getEntity(ctx, req, "snapshots", &sps)
+}
+
+// GetSnapshot retrieves a single snapshot for a specific ID and returns an error if any.
+func (c *Client) GetSnapshot(ctx context.Context, idr string, req *Request) (*Snapshot, error) {
+	snp := new(Snapshot)
+	return snp, c.getEntity(ctx, req, fmt.Sprintf("snapshots/%s", idr), snp)
+}
+
+// HeadSnapshot retrieves only the headers of a single snapshot for a specific ID, and returns an error if any.
+func (c *Client) HeadSnapshot(ctx context.Context, idr string) (*Headers, error) {
+	return c.headEntity(ctx, fmt.Sprintf("snapshots/%s/download", idr))
+}
+
+// ReadSnapshot reads the contents of a single snapshots for a specific ID, and invokes the specified callback function for each chunk read.
+func (c *Client) ReadSnapshot(ctx context.Context, idr string, cbk ReadCallback) error {
+	return c.readEntity(ctx, fmt.Sprintf("snapshots/%s/download", idr), cbk)
+}
+
+// DownloadSnapshot downloads the contents of a single snapshot for a specific ID, and writes the data to the specified WriteSeeker.
+func (c *Client) DownloadSnapshot(ctx context.Context, idr string, wsk io.WriteSeeker) error {
+	return c.downloadEntity(ctx, fmt.Sprintf("snapshots/%s/download", idr), wsk)
+}
+
+// GetArticles retrieves articles from the API based on the given name and request parameters.
+func (c *Client) GetArticles(ctx context.Context, nme string, req *Request) ([]*Article, error) {
+	ats := []*Article{}
+	return ats, c.getEntity(ctx, req, fmt.Sprintf("articles/%s", nme), &ats)
+}
+
+// GetStructuredContents retrieves structured contents from the API based on the given name and request parameters.
+func (c *Client) GetStructuredContents(ctx context.Context, nme string, req *Request) ([]*StructuredContent, error) {
+	ats := []*StructuredContent{}
+	return ats, c.getEntity(ctx, req, fmt.Sprintf("structured-contents/%s", nme), &ats)
+}
+
+// StreamArticles streams all available articles from the server and applies a callback function to each article
+// as they arrive. The callback function must implement the ReadCallback interface.
+func (c *Client) StreamArticles(ctx context.Context, req *Request, cbk ReadCallback) error {
+	return c.subscribeToEntity(ctx, "articles", req, cbk)
+}
+
+// ReadAll reads the contents of the given io.Reader and calls the given ReadCallback function
+// with each chunk of data read.
+func (c *Client) ReadAll(ctx context.Context, rdr io.Reader, cbk ReadCallback) error {
+	return c.readAll(ctx, rdr, cbk)
```

Comment:
`readAll` private method seems redundant, if we are going to make a public wrapper `ReadAll` afterall.

Reply:
Good point, removed the private method and made it public.